assign2/src/view/account/UserAccount/SmartUserAccountView.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import User from "../../../model/objects/User";
+import {AppState} from "../../../model/Model";
+import {Dispatch} from "redux";
+import {connect} from "react-redux";
+import {getPostsByAuthorId} from "../../../model/question/selectors";
+import {RouteComponentProps} from "react-router";
+import {getUserById} from "../../../model/user/selectors";
+import Answer from "../../../model/objects/Answer";
+import Question from "../../../model/objects/Question";
+import UserAccountView from "./UserAccountView";
+import {AnswerView} from "../../general/reuseable/AnswerView";
+import {QuestionView} from "../../general/reuseable/QuestionView"
+import Post from "../../../model/objects/Post";
+import {userAccountPresenter} from "../../../presesnter/UserAccountPresenter";
+
+interface Props {
+    posts: Post[];
+    currentUser?: User;
+    viewedProfile?: User;
+    onQuestionInputChange: (questionId: number) => (value: string) => void;
+    onAnswerInputChange: (answerId: number) => (value: string) => void;
+    onUpdateQuestion: (questionId: number) => () => void;
+    onUpdateAnswer: (answerId: number) => () => void;
+    onDeleteQuestion: (questionId: number) => () => void;
+    onDeleteAnswer: (answerId: number) => () => void;
+}
+
+class SmartUserAccountView extends React.Component<Props> {
```

Comment:
Hmm, how come you are always extending the Component base class even for "dumb" components? Check out: https://medium.com/@ethan_ikt/react-stateless-functional-component-with-typescript-ce5043466011